crates/cast/src/starknet_commands/script.rs

```diff
@@ -0,0 +1,341 @@
+                            .map(StarknetConversions::to_felt252),
+                    )
+                    .expect("Failed to insert data");
+
+                Ok(())
+            }
+            _ => Err(anyhow!("Unknown cheatcode selector: {selector}")),
+        }?;
+
+        let result_end = buffer.ptr;
+        insert_value_to_cellref!(vm, output_start, result_start)?;
+        insert_value_to_cellref!(vm, output_end, result_end)?;
+
+        Ok(())
+    }
+}
+
+pub fn run(
+    module_name: &str,
+    path_to_scarb_toml: &Option<Utf8PathBuf>,
+    provider: &JsonRpcClient<HttpTransport>,
+    runtime: Runtime,
+) -> Result<ScriptResponse> {
+    let path = compile_script(path_to_scarb_toml.clone())?;
+
+    let sierra_program = serde_json::from_str::<VersionedProgram>(
+        &fs::read_to_string(path.clone())
+            .with_context(|| format!("failed to read Sierra file: {path}"))?,
+    )
+    .with_context(|| format!("failed to deserialize Sierra program: {path}"))?
+    .into_v1()
+    .with_context(|| format!("failed to load Sierra program: {path}"))?;
+
+    let runner = SierraCasmRunner::new(
+        sierra_program,
+        Some(MetadataComputationConfig::default()),
+        OrderedHashMap::default(),
+    )
+    .with_context(|| "Failed setting up runner.")?;
+
+    let name_suffix = module_name.to_string() + "::main";
+    let func = runner.find_function(name_suffix.as_str())?;
+
+    let (entry_code, builtins) = runner.create_entry_code(func, &Vec::new(), usize::MAX)?;
+    let footer = runner.create_code_footer();
+    let instructions = chain!(
+        entry_code.iter(),
+        runner.get_casm_program().instructions.iter(),
+        footer.iter()
+    );
+
+    // import from cairo-lang-runner
+    let (hints_dict, string_to_hint) = build_hints_dict(instructions.clone());
+
+    let mut cairo_hint_processor = CairoHintProcessor {
+        hints: &string_to_hint,
+        provider,
+        runtime,
+        run_resources: RunResources::default(),
+    };
+
+    match runner.run_function(
+        func,
+        &mut cairo_hint_processor,
+        hints_dict,
+        instructions,
+        builtins,
+    ) {
+        Ok(result) => match result.value {
+            RunResultValue::Success(data) => Ok(ScriptResponse {
+                status: "success".to_string(),
+                msg: build_readable_text(&data),
+            }),
+            RunResultValue::Panic(panic_data) => Ok(ScriptResponse {
+                status: "script panicked".to_string(),
+                msg: build_readable_text(&panic_data),
+            }),
+        },
+        Err(err) => Err(err.into()),
+    }
+}
+
+fn compile_script(path_to_scarb_toml: Option<Utf8PathBuf>) -> Result<Utf8PathBuf> {
+    let scripts_manifest_path = match path_to_scarb_toml {
+        Some(path) => path,
+        None => get_scarb_manifest()
+            .context("Failed to obtain manifest path from scarb")
+            .unwrap(),
+    };
```

Comment:
Not sure if this logic should be here. It would make sense to accept `scripts_manifest_path` as `Utf8PathBuf` here and do the optional handling at caller site

Reply:
we'll think about it after release, since this is non blocking. thanks.